### ISSUE TITLE
Skip builder boost factor calculation when the default value is provided

### ISSUE THREAD

**Issue by eserilev** [sigp/lighthouse#5343]
## Description

If the default `builder_boost_factor` value  is provided in a `validator/blocks` v3  request, the builder payload value is calculated with the boost factor. This causes some unnecessary loss of information, e.g. 

```
INFO Local block is more profitable than relay block, 
builder_boost_factor: Some(100), 
boosted_relay_value: 3745718458417800, 
relay_value: 3745718458417893,
local_block_value: 3762801154811571, service: exec
```

We should instead skip the boost factor calculation when the default value is provided

**Comment by pahor167** [comment]
I created [PR](https://github.com/sigp/lighthouse/pull/5352)

**Comment by jimmygchen** [comment]
Fixed in #5352, thanks @pahor167!
